lib/bap_image/llvm_binary_stubs.hpp

```diff
@@ -331,12 +331,35 @@ uint64_t image_entry(const MachOObjectFile* obj) {
 }
 
 uint64_t image_entry(const COFFObjectFile* obj) {
-    const pe32_header* hdr = 0;
-    if (error_code ec = obj -> getPE32Header(hdr))
-        llvm_binary_fail(ec);
-    if (!hdr)
-        llvm_binary_fail("PE header not found");
-    return hdr->AddressOfEntryPoint;
+    if (obj->getBytesInAddress() == 4) {
+        const pe32_header* hdr = 0;
+        if (error_code ec = obj -> getPE32Header(hdr))
+            llvm_binary_fail(ec);
+        if (!hdr)
+            llvm_binary_fail("PE header not found");
+        return hdr->AddressOfEntryPoint;
+    } else {        
+        // llvm version 3.4 doesn't support pe32plus_header,
+        // but in version 3.5 it does. So, later one will be 
+        // able to write obj->getPE32PlusHeader(hdr) for 64-bit files.
+        uint64_t cur_ptr = 0;
+        StringRef data = obj->getData();
+        MemoryBuffer *buf = MemoryBuffer::getMemBuffer(data, "", false);
```

Comment:
1. memory leak
2. return value is not checked
3. why are you creating this buffer at all, if you only need a pointer to the start, that you already have as it is equal to `data`